### PR TITLE
Introduce OpSignature to IR

### PR DESCRIPTION
**Pull request by justinchuby**
Introduce OpSignature accessible from the `.op_signature` property of all OpLike objects (traced function, onnx function and op). The OpSignature class leverages the IR to represent the signature of an operator, preserving ordering of all inputs and provides easy to work with type representations.

The PR also deprecates the ParamSchema class and properties.

Fixes https://github.com/microsoft/onnxscript/issues/1697

The next PR will replace param_schemas usage.

### CODE REVIEW

File: onnxscript/ir/_schemas.py [hunk 1]
```diff
@@ -0,0 +1,544 @@
...
+def _get_type_from_str(
+    type_str: str,
+) -> ir.TensorType | ir.SequenceType | ir.OptionalType:
+    """Converter a type_str from ONNX Opschema to ir.TypeProtocol.
+
+    A type str has the form of "tensor(float)" or composite type like "seq(tensor(float))".
+    """
+
+    # TODO: Upstream this to IR
+
```
Comment (justinchuby) [inline onnxscript/ir/_schemas.py:156]:
```suggestion
    """
```

File: onnxscript/ir/_schemas.py [hunk 2]
```diff
@@ -0,0 +1,548 @@
+# Copyright (c) Microsoft Corporation.
```
Comment (titaiwangms) [inline onnxscript/ir/_schemas.py:1]: Should we put a comment saying this should align to torch _schema.py until we move it to here?

Comment (titaiwangms) [reply onnxscript/ir/_schemas.py:1]: Actually, we only have the test for this file, which means that torch's _schema.py should be following this one.

**Review by titaiwangms — APPROVED** [review]

**Comment by justinchuby** [comment]
When the signature property is defined, PyTorch onnx exporter will not generate a new signature, and instead use what is here. However, since it is doing isinstance checks on the signature the exporter itself defines, we get isinstance errors there.

**Pull request merged**